core/src/main/java/io/grpc/internal/TransportSet2.java

```diff
@@ -80,7 +79,9 @@
   private final Callback callback;
   private final ClientTransportFactory transportFactory;
   private final ScheduledExecutorService scheduledExecutor;
-  private final Executor appExecutor;
+
+  // A serializing executor shared across the Channel
+  private final Executor channelExecutor;
```

Comment:
Don't we want to be more strict here and type it `SerializingExecutor` ? (and in constructor accordingly)